.github/workflows/mac.yml

```diff
@@ -0,0 +1,55 @@
+name: 'Build for macOS'
+on: [push, pull_request]
+
+jobs:
+    build:
+      runs-on: ${{ matrix.os }}
+      strategy:
+
+          matrix:
+              os: [macos-latest]
+              architecture: [x64, arm64]
+              node-version: [20.x]
+      steps:
```

Comment:
From the GitHub documentation only `macos-latest` or `macos-14` is with Apple Silicon M1 (arm64) to compile for Intel (x86_x64) we should use the runner `macos-12` or `macos-13`

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

https://github.com/actions/runner-images?tab=readme-ov-file#available-images